zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java

```diff
@@ -213,7 +214,7 @@ static HttpData convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage
     String encoding = request.headers().get(HttpHeaderNames.CONTENT_ENCODING);
     HttpData content = request.content();
     if (!content.isEmpty() && encoding != null && encoding.contains("gzip")) {
-      content = GZIP_DECODER_FACTORY.newDecoder().decode(content);
+      content = GZIP_DECODER_FACTORY.newDecoder(ByteBufAllocator.DEFAULT).decode(content);
```

Comment:
@anuraaga @trustin this was a guess. Is it correct to do this or is there a request-specific means to choose the most relevant byte buff allocator?

Reply:
You can use ctx.alloc() 😄

Reply:
nice figured it was something like that!